components/forms/Form/Form.tsx

```diff
@@ -45,6 +46,20 @@ const InnerForm = (props: InnerFormProps & FormikProps<FormValues>) => {
     }
   }, [formStatusError, errorList, lastSubmitCount, canFocusOnError]);
 
+  useEffect(() => {
+    // timeout to prevent form from being submitted too quickly
+    const millisPerFormElement = 2000;
+    const formSubmitTimeout = setTimeout(
+      () => {
+        setFormSubmitDelay(true);
+      },
+      children ? React.Children.count(children) * millisPerFormElement : 0
+    );
+    return () => {
+      clearTimeout(formSubmitTimeout);
+    };
+  }, [formSubmitDelay]);
```

Comment:
With `formSubmitDelay` added as a useEffect() dependency it will actionally run 1 more time then you might expect.
- Inital form render, useEffect runs and create setTimeout callback, formSubmitDelay state = false
- Timer expires and executes callback, formSubimtDelay state = true
- UseEffect dependency detected and checks if previous value = new value.  false = true so run useEffect again.
- useEffect runs and creates setTimeout callback, formSubmitDelay state = true
- Timer expires and executre callback, formSubmitDelay state = true
- UseEffect dependency detected and checks if previous value = new value.  true = true so do not run useEffect again.

This behavior can be avoided by instead passing an empty dependnecy array `useEffect(()=> (), [] )`
An empty dependency array tells useEffect to only run on the initial render and never again.

For more info: https://reactjs.org/docs/hooks-effect.html#tip-optimizing-performance-by-skipping-effects